<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>760</width>
    <height>570</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <widget class="GLWidget" name="glWidget" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>630</width>
     <height>551</height>
    </rect>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>360</y>
     <width>100</width>
     <height>100</height>
    </rect>
   </property>
   <property name="title">
    <string>Смещение</string>
   </property>
   <widget class="QWidget" name="formLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>81</width>
      <height>74</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>X:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="XTransl">
       <property name="minimum">
        <number>-10</number>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="YTransl">
       <property name="minimum">
        <number>-10</number>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSpinBox" name="ZTransl">
       <property name="minimum">
        <number>-10</number>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Y:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Z:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>460</y>
     <width>100</width>
     <height>100</height>
    </rect>
   </property>
   <property name="title">
    <string>Вращение</string>
   </property>
   <widget class="QWidget" name="formLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>81</width>
      <height>74</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>X:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="XRot">
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="YRot">
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSpinBox" name="ZRot">
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Y:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Z:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_4">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>270</y>
     <width>100</width>
     <height>81</height>
    </rect>
   </property>
   <property name="title">
    <string>Косоугольная пр.</string>
   </property>
   <widget class="QWidget" name="formLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>81</width>
      <height>48</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout_3">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>a:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="projectionA">
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>90</number>
       </property>
       <property name="value">
        <number>45</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>f:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QDoubleSpinBox" name="projectionF">
       <property name="minimum">
        <double>-5.000000000000000</double>
       </property>
       <property name="maximum">
        <double>5.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.050000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_5">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>10</y>
     <width>100</width>
     <height>181</height>
    </rect>
   </property>
   <property name="title">
    <string>Объект</string>
   </property>
   <widget class="QRadioButton" name="surface_1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>82</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Тело 1</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="surface_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>82</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Тело 2</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Коеф. параболы:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="curvePoints">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Точки кривой</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="parabParam">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="decimals">
     <number>1</number>
    </property>
    <property name="minimum">
     <double>-5.000000000000000</double>
    </property>
    <property name="maximum">
     <double>5.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QCheckBox" name="showNormals">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>нормали</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="shading">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>закраска</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_6">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>200</y>
     <width>100</width>
     <height>61</height>
    </rect>
   </property>
   <property name="title">
    <string>Проекция</string>
   </property>
   <widget class="QRadioButton" name="orthoProjection">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>82</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>косоугол-я</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="surface_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>82</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>ортогонал-я</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
   <slots>
    <slot>translateX(int)</slot>
    <slot>translateY(int)</slot>
    <slot>translateZ(int)</slot>
    <slot>rotateX(int)</slot>
    <slot>rotateY(int)</slot>
    <slot>rotateZ(int)</slot>
    <slot>projectionA(int)</slot>
    <slot>projectionF(double)</slot>
    <slot>parabolaParam(double)</slot>
    <slot>changeSurface(bool)</slot>
    <slot>showNormalsForFace(bool)</slot>
    <slot>editCurve()</slot>
    <slot>changeProjection(bool)</slot>
    <slot>setShadind(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>XTransl</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>translateX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>YTransl</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>translateY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ZTransl</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>translateZ(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>XRot</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>rotateX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>YRot</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>rotateY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ZRot</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>rotateZ(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>202</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>projectionA</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>projectionA(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>projectionF</sender>
   <signal>valueChanged(double)</signal>
   <receiver>glWidget</receiver>
   <slot>projectionF(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>surface_1</sender>
   <signal>toggled(bool)</signal>
   <receiver>glWidget</receiver>
   <slot>changeSurface(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>700</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>parabParam</sender>
   <signal>valueChanged(double)</signal>
   <receiver>glWidget</receiver>
   <slot>parabolaParam(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showNormals</sender>
   <signal>toggled(bool)</signal>
   <receiver>glWidget</receiver>
   <slot>showNormalsForFace(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>694</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>curvePoints</sender>
   <signal>clicked()</signal>
   <receiver>glWidget</receiver>
   <slot>editCurve()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>700</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>orthoProjection</sender>
   <signal>toggled(bool)</signal>
   <receiver>glWidget</receiver>
   <slot>changeProjection(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>700</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shading</sender>
   <signal>toggled(bool)</signal>
   <receiver>glWidget</receiver>
   <slot>setShadind(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>694</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>285</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
